src/synapse/azext_synapse/_params.py

```diff
@@ -3,159 +3,204 @@
         c.argument('max_node_count', arg_group='AutoScale', help='The max node count.')
         c.argument('min_node_count', arg_group='AutoScale', help='The min node count.')
 
         # AutoPause
-        c.argument('auto_pause_enabled', arg_type=get_three_state_flag(), arg_group='AutoPause',
+        c.argument('enable_auto_pause', arg_type=get_three_state_flag(), arg_group='AutoPause',
                    help='The flag of enabling auto pause.')
         c.argument('delay_in_minutes', arg_group='AutoPause', help='The delay time.')
 
         # Environment Configuration
-        c.argument('library_requirements_content', arg_group='Environment Configuration',
-                   help='The library requirements content.')
-        c.argument('library_requirements_filename', arg_group='Environment Configuration',
-                   help='The library requirements file name.')
+        c.argument('library_requirements_file', arg_group='Environment Configuration',
+                   help='The library requirements file.')
 
         # Default Folder
-        c.argument('spark_events_folder', arg_group='Default Folder', help='The spark events folder.')
-        c.argument('default_spark_log_folder', arg_group='Default Folder', help='The default spark log folder.')
+        c.argument('spark_events_folder', arg_group='Default Folder', help='The Spark events folder.')
+        c.argument('default_spark_log_folder', arg_group='Default Folder', help='The default Spark log folder.')
 
         # Component Version
-        c.argument('spark_version', arg_group='Component Version', help='The supported spark version is 2.4 now.')
+        c.argument('spark_version', arg_group='Component Version', help='The supported Spark version is 2.4 now.')
 
-        c.argument('force', help='The flag of force operation.')
         c.argument('tags', arg_type=tags_type)
 
     with self.argument_context('synapse spark pool update') as c:
         c.argument('tags', arg_type=tags_type)
+        # Node
+        c.argument('node_count', arg_group='Node', help='The number of node.')
+        c.argument('node_size_family', arg_group='Node', help='The node size family.')
+
+        c.argument('node_size', arg_group='Node', arg_type=get_enum_type(['Small', 'Medium', 'Large']),
+                   help='The node size.')
+        # AutoScale
+        c.argument('enable_auto_scale', arg_type=get_three_state_flag(), arg_group='AutoScale',
+                   help='The flag of enabling auto scale.')
+        c.argument('max_node_count', arg_group='AutoScale', help='The max node count.')
+        c.argument('min_node_count', arg_group='AutoScale', help='The min node count.')
+
+        # AutoPause
+        c.argument('enable_auto_pause', arg_type=get_three_state_flag(), arg_group='AutoPause',
+                   help='The flag of enabling auto pause.')
+        c.argument('delay_in_minutes', arg_group='AutoPause', help='The delay time.')

```

Comment:
Don't put units in argument names, and always put the expected units in the help text. use "delay" instead.

Reply:
fixed.

---
In reply to: [424950273](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r424950273) [](ancestors = 424950273)